### PR TITLE
fix(vitest): split out tests

### PR DESCRIPTION
**Pull request by kanadgupta**
## 🧰 Changes

This PR is branched off of #857 and splits out our test bed into two parts: single-threaded and multi-threaded. The tests that require different directory changes (i.e. they use `process.chdir()` have been moved into the former, while the latter contains the vast majority of all tests.

For the single-threaded tests, [MSW](https://mswjs.io/) has been introduced because `nock` wasn't working properly when attempting to parse and validate certain request bodies. The goal is to eventually introduce MSW across the entire repository, but that's not a priority at the moment.

## 🧬 QA & Testing

Do tests pass?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,8 +38,8 @@ jobs:
       - name: Build dist
         run: npm run build
 
-      - name: Run tests
-        run: npm test
+      - name: Run tests (${{ matrix.thread }}-threaded)
+        run: npm run test:${{ matrix.thread }}
```
Comment (kanadgupta) [inline .github/workflows/ci.yml:42]: this was more of an experiment — seems like a lot of unnecessary build overhead with basically zero time savings — feel free to revert!

File: __tests__/helpers/get-gha-setup.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import * as getPkgVersion from '../../src/lib/getPkgVersion';
 
 import getGitRemoteMock from './get-git-mock';
 
-const testWorkingDir = process.cwd();
+// const testWorkingDir = process.cwd();
```
Comment (erunion) [inline __tests__/helpers/get-gha-setup.ts:13]:
```suggestion
```

File: __tests__/single-threaded/openapi/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+      //   .reply(201, { registryUUID, spec: { openapi: '3.0.0' } });
+
+      // const mockWithHeader = getAPIMockWithVersionHeader(version)
+      //   .get('/api/v1/api-specification')
+      //   .basicAuth({ user: key })
+      //   .reply(200, [])
+      //   .post('/api/v1/api-specification', { registryUUID })
+      //   .delayConnection(1000)
+      //   .basicAuth({ user: key })
+      //   .reply(201, { _id: 1 }, { location: exampleRefLocation });
```
Comment (erunion) [inline __tests__/single-threaded/openapi/index.test.ts:88]: some left in mocking in here

**Review by erunion — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kanadgupta) [reply .github/workflows/ci.yml]: update: reverted in https://github.com/readmeio/rdme/pull/870/commits/a870661f83ad789e4d956728019a18dd743f2622

**Pull request merged**